content/client-lib-development-guide/features.textile

```diff
@@ -630,6 +630,10 @@ h3(#realtime-channel). Channel
 ** @(RTL19c)@ In the case of a delta message with a @vcdiff@ @encoding@ step, the @vcdiff@ decoder must be used to decode the base payload of the of delta message, applying that delta to the stored base payload. The direct result of that vcdiff delta application, before performing any further decoding steps, is stored as the updated base payload.
 * @(RTL20)@ The @id@ of the last received message on each channel must be stored along with the base payload. When processing a delta message (i.e. one whose @encoding@ contains @vcdiff@ step) the stored last message @id@ must be compared against the delta reference @id@, indicated in the @Message.extras.delta.from@ field of the delta message. If the delta reference @id@ of the received delta message does not equal the stored @id@ corresponding to the base payload, the message decoding must fail. The recovery procedure from "RTL18":#RTL18 must be executed.
 * @(RTL21)@ The messages in the @messages@ array of a @ProtocolMessage@ should each be decoded in ascending order of their index in the array.
+* @(RTL22)@ Methods must be provided for creating and destroying a listener which only fires when the message matches a set of criteria
+** @(RTL22a)@ The method must allow for filters matching one or more of: @extras.reference.id@, @extras.reference.type@ or @name@
+** @(RTL22b)@ The method must allow for matching only messages which contain no @extras.reference.type@
```

Comment:
This requirement seems a bit odd, why would a reference not have a type, and why would I want to filter specifically for references which have no type?

Reply:
[Done](https://github.com/ably/docs/pull/1444/commits/5b4d8a9b652d63d6bbc388ee022a3ac6dceebbab). This was just a mistype as the way it was always meant to be implemented was to check for the presence of a reference.